app/controllers/concerns/bbb_server.rb

```diff
@@ -88,12 +83,19 @@ def start_session(room, options = {})
       else
         bbb_server.create_meeting(room.name, room.bbb_id, create_options)
       end
-
-      unless meeting[:messageKey] == 'duplicateWarning'
-        room.update_attributes(sessions: room.sessions + 1, last_session: DateTime.strptime(meeting[:createTime].to_s, "%Q"))
+      logger.warn "Duplicated create request" if meeting[:messageKey] == 'duplicateWarning'
```

Comment:
If it's a diplicate create request, shouldn't we `return logger.warn ... ` so that the rest of the code doesn't run?

Reply:
IMHO we cannot consider a duplicated request as a problem on GL and here's why:
We'll still want the start session logic to complete successfully (with no exceptions raised and even on duplicated requests) and with the inconsistency resolved to further complete the **joining** (`join_path` always calls `start_session` before returning a `join_meeting_url`) pipeline.
Halting the session because of duplication will prevent the session starter from joining it resulting in:
1. An error on their side (For no reason I believe, since duplicated requests doesn't indicate a problem on BBB).
2. **The room won't be declared as running** even though the session started on BBB because it had no attendee.
The room session starter will still have to start it resulting in the same error causing a `dead lock` on the room state until its session gets explicitly or implicitly ended meaning that whenever an issue raises causing this inconsistency we'll still face the problem (In fact it would be worst because this time there won't be errors on client side).
A duplicated request doesn't mean an error it only indicates that such request has been made i.e. two users attempt to start the same session BBB won't consider it an error and one if precedes the other will create the session while the other will simply make a duplicate request that will return details about the meeting as expected but with an indication if useful that this isn't this request was only passive.
Considering duplication as an error in such cases won't create a `dead lock` but it would definitely give a bad UX for those who observe a flash message or gets redirected when they were able to join the session despite their duplicate request just like BBB API encourage it.

Reply:
https://docs.bigbluebutton.org/dev/api.html#create

`The create call is idempotent: you can call it multiple times with the same parameters without side effects.`

So the `duplicateWarning` is safe to ignore.

Reply:
Exactly,
Thanks @GhaziTriki ,I've been looking for the word **idempotent** it could have helped simplify my explanation xD
The useless updates to the room `last_session` and `sessions` attributes that we should avoid aren't those with the duplicated requests but the ones in which the room attributes is already consistent and that is this PR's solution in a nutshell.